File: code/frontend/vue3/src/modules/site/types/blog.ts
// src/modules/site/types/blog.ts —— 博客文章类型

/** 发布来源: markdown 在线编辑 / url 关联外链 */
export type PostSource = 'markdown' | 'url'

/** 发布状态: 草稿 / 已发布 */
export type PostStatus = 'draft' | 'published'

/** 博客文章实体 */
export interface BlogPost {
  id: string
  /** 作者用户ID */
  user_id: string
  title: string
  source_type: PostSource
  /** markdown 正文(source_type=url 时可空) */
  content: string
  /** 关联外链地址(source_type=url 时使用) */
  url: string | null
  category: string | null
  status: PostStatus
  created_at: string
  updated_at: string
}

/** 创建博客文章参数 */
export interface BlogPostCreate {
  title: string
  source_type?: PostSource
  content?: string
  url?: string | null
  category?: string | null
  status?: PostStatus
}

/** 更新博客文章参数(全部可选, 仅更新传入字段) */
export type BlogPostUpdate = Partial<BlogPostCreate>

File: code/backend/server_py/src/module_site/service/blog.py
from common.utils.db.schema.pagination import PaginationParams, PaginationResponse
from module_site.do.blog import BlogPost, BlogPostCreate, BlogPostUpdate, PostStatus
from module_site.dao.blog import BlogPostDao


class BlogPostService:
    """博客文章服务: 发布(在线 markdown/关联 URL)、管理与展示"""

    def __init__(self, blog_post_dao: BlogPostDao):
        """依赖注入构造器: 初始化所需的数据访问对象"""
        self.blog_post_dao = blog_post_dao

    async def add(self, post: BlogPostCreate, user_id: str) -> str:
        """新增文章(归属当前用户)
        :return: 新建文章ID
        """
        return await self.blog_post_dao.add(post, user_id)

    async def get_visible(self, post_id: str, user_id: str) -> BlogPost | None:
        """查询单篇文章(带可见性判定)
        已发布文章对持有 read 权限的用户可见; 草稿仅作者可见
        :param user_id: 请求者用户ID
        """
        result = await self.blog_post_dao.get(post_id)
        if not result:
            return None
        if result.user_id != user_id and result.status != PostStatus.Published:
            return None
        return result

    async def update(self, post_id: str, post: BlogPostUpdate, user_id: str) -> None:
        """更新本人文章"""
        await self.blog_post_dao.update(post_id, post, user_id)

    async def delete(self, post_id: str, user_id: str) -> None:
        """删除本人文章"""
        await self.blog_post_dao.delete(post_id, user_id)

    async def list_mine(
        self,
        pagination: PaginationParams,
        user_id: str,
        title: str | None = None,
        status: str | None = None,
        source_type: str | None = None,
    ) -> PaginationResponse:
        """分页获取本人文章列表(管理页)"""
        items = await self.blog_post_dao.list_mine(
            pagination, user_id, title=title, status=status, source_type=source_type
        )
        total = await self.blog_post_dao.count_mine(
            user_id, title=title, status=status, source_type=source_type
        )
        return PaginationResponse.create(items, total, pagination)

    async def list_published(
        self,
        pagination: PaginationParams,
        keyword: str | None = None,
    ) -> PaginationResponse:
        """分页获取已发布文章列表(展示页)"""
        items = await self.blog_post_dao.list_published(pagination, keyword=keyword)
        total = await self.blog_post_dao.count_published(keyword=keyword)
        return PaginationResponse.create(items, total, pagination)
